tests/resources/test_upload_jobs.py

```diff
@@ -975,21 +975,35 @@ def test_extra_metadata(cidc_api, clean_db, monkeypatch):
     assert res.status_code == 400
     assert "files" in res.json["_error"]["message"]
 
-    merge_extra_metadata = MagicMock()
-    monkeypatch.setattr(
-        "cidc_api.models.UploadJobs.merge_extra_metadata", merge_extra_metadata
+    with open("tests/resources/data/npx_valid.xlsx", "rb") as f:
+        valid_npx = f.read()
+    with open("tests/resources/data/npx_invalid.xlsx", "rb") as f:
+        invalid_npx = f.read()
+
+    res = client.post(
+        "/ingestion/extra-assay-metadata",
+        data={"job_id": 123, "uuid-1": (io.BytesIO(valid_npx), "fname1")},
     )
+    assert res.status_code == 400
+    assert "123" in res.json["_error"]["message"]
 
+    job_id, _ = setup_upload_jobs(cidc_api)
     res = client.post(
         "/ingestion/extra-assay-metadata",
         data={
-            "job_id": 123,
-            "uuid-1": (io.BytesIO(b"fake file 1"), "fname1"),
-            "uuid-2": (io.BytesIO(b"fake file 2"), "fname2"),
+            "job_id": job_id,
+            "uuid-1": form_data("olink.xlsx", io.BytesIO(valid_npx), "olink"),
         },
     )
-    assert res.status_code == 200
-    merge_extra_metadata.assert_called()
+    assert res.status_code == 400
+    assert "files in request" in res.json["_error"]["message"]
+
+    # merge_extra_metadata = MagicMock()
+    # monkeypatch.setattr(
+    #     "cidc_api.models.UploadJobs.merge_extra_metadata", merge_extra_metadata
+    # )
+    # TODO add test that actually passes
+    # merge_extra_metadata.assert_called()
```

Comment:
yeah, so I'm only unsure about this here

Reply:
I couldn't figure out how to make it run a test that would come back with a 200, I think I'll have to get/generate actual olink data

Reply:
fixed by commit 3852882